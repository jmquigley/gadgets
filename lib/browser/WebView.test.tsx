"use strict";

import {mount} from "enzyme";
import assert from "power-assert";
import * as React from "react";
import {getDefaultWebViewProps, getDefaultWebViewState, WebView} from "./index";

test("Test retrieval of the WebView props object", () => {
	const props = getDefaultWebViewProps();

	assert(props);
	expect(props).toMatchSnapshot();

	const state = getDefaultWebViewState();

	assert(state);
	expect(state).toMatchSnapshot();
});

test("Test creation of a WebView instance", () => {
	const ctl = mount(<WebView />);

	assert(ctl);
	expect(ctl).toMatchSnapshot();
});
