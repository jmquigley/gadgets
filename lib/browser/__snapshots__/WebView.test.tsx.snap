// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test creation of a WebView instance 1`] = `
<WebView
  controlled={true}
  direction="right"
  disabled={false}
  focus={false}
  hidden={false}
  innerRef={[Function]}
  location="none"
  noborder={false}
  noedit={false}
  nohover={false}
  nopropagation={false}
  notheme={false}
  notooltip={false}
  obj="WebView"
  ripple={false}
  selected={false}
  sizing="normal"
  style={Object {}}
  testing={true}
  visible={true}
/>
`;

exports[`Test retrieval of the WebView props object 1`] = `
Object {
  "controlled": true,
  "direction": "right",
  "disabled": false,
  "focus": false,
  "hidden": false,
  "innerRef": [Function],
  "location": "none",
  "noborder": false,
  "noedit": false,
  "nohover": false,
  "nopropagation": false,
  "notheme": false,
  "notooltip": false,
  "obj": "WebView",
  "ripple": false,
  "selected": false,
  "sizing": "normal",
  "style": Object {},
  "testing": true,
  "visible": true,
}
`;

exports[`Test retrieval of the WebView props object 2`] = `
Object {
  "children": null,
  "sizing": "normal",
  "style": Object {},
}
`;
